src/main/kotlin/chessdt/chesswithdyegoandtobi/HelloController.kt

```diff
@@ -1,44 +1,54 @@
 package chessdt.chesswithdyegoandtobi
 
+import chessdt.chesswithdyegoandtobi.domain.MoveArguments
 import com.github.bhlangonijr.chesslib.Board
 import com.github.bhlangonijr.chesslib.Square
 import com.github.bhlangonijr.chesslib.move.Move
 import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
 import org.springframework.web.bind.annotation.RestController
 
 @RestController
 class HelloController {
 
+    // create new board
+    val board = Board()
+
     @GetMapping("/hello")
     fun sayHelloTo(name: String): String {
         return "HELLO DYEGO! $name"
     }
 
     @GetMapping("/moves/getLegal")
     fun getLegalMoves(currentFigure: String): List<Move> {
-        val board = Board()
         val allLegalMoves = board.legalMoves()
-        val legalMovesForCurrentFigure = allLegalMoves.filter { it.toString().startsWith(currentFigure) }
-
-        return legalMovesForCurrentFigure
+        return allLegalMoves.filter { it.toString().startsWith(currentFigure.lowercase()) }
     }
 
-    @GetMapping("/chessMagic")
-    fun chessMagic(): String {
-        // Creates a new chessboard in the standard initial position
-        // Creates a new chessboard in the standard initial position
-        val board = Board()
+    @PostMapping("/chessMagic")
+    fun chessMagic(
+        @RequestBody moveArguments: MoveArguments
+    ): String {
+        val startPosition = moveArguments.startPosition
+        val endPosition = moveArguments.endPosition
 
-        // Make a move from E2 to E4 squares
-        board.doMove(Move(Square.E2, Square.E4))
+        val startSquare = Square.valueOf(startPosition)
+        val endSquare = Square.valueOf(endPosition)
 
-        board.doMove(Move(Square.B7, Square.B4))
+        // check if move is legal
+        val legalMoves = getLegalMoves(startPosition)
+        val selectedMove = Move(startSquare, endSquare)
+        val isSelectedMoveLegal = selectedMove in legalMoves
 
-        // print the chessboard in a human-readable form
+        // Make a move if legal
+        if (isSelectedMoveLegal) {
+            board.doMove(Move(startSquare, endSquare))
+        } else {
+            print("Fuck Off")
+        }
 
-        val newBoard = board.toString()
         // print the chessboard in a human-readable form
-
-        return newBoard
+        return board.toString()
     }
```

Comment:
Niceee Tobi, that is it! 🚀 

Just to show you about [Kotlin extensions](https://kotlinlang.org/docs/extensions.html) and about moving the business logic to the domain, this would be an alternative implementation
```
    @PostMapping("/chessMagic")
    fun chessMagic(
        @RequestBody moveArguments: MoveArguments
    ): String {
        val startSquare = Square.valueOf(moveArguments.startPosition)
        val endSquare = Square.valueOf(moveArguments.endPosition)

        // check if move is legal
        val legalMoves = getLegalMoves(moveArguments.startPosition)
        val selectedMove = Move(startSquare, endSquare)

        // Make a move if legal
        if (selectedMove.isLegal(legalMoves)) {
            board.doMove(Move(startSquare, endSquare))
        } else {
            print("Fuck Off")
        }

        // print the chessboard in a human-readable form
        return board.toString()
    }

    fun Move.isLegal(legalMoves: List<Move>) = this in legalMoves
```